packages/contracts/src/contracts/current/protocol/Exchange/MixinExchangeCore.sol

```diff
@@ -106,10 +110,10 @@ contract MixinExchangeCore is
         }
 
         // Validate order availability
-        uint256 remainingTakerTokenAmount = safeSub(order.takerTokenAmount, getUnavailableTakerTokenAmount(orderHash));
+        uint256 remainingTakerTokenAmount = safeSub(order.takerTokenAmount, filled[orderHash]);
         takerTokenFilledAmount = min256(takerTokenFillAmount, remainingTakerTokenAmount);
         if (takerTokenFilledAmount == 0) {
```

Comment:
If we change this to `remainingTakerTokenAmount == 0` we no don't depend on `require(takerTokenFillAmount > 0)` to keep this condition correct.